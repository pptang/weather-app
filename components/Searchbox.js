// @flow
import * as React from 'react';

type Props = {
  onSearch: Function,
};

type State = {
  location: string,
};

class Searchbox extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      location: '',
    };
  }
  render() {
    return (
      <div className="searchboxContainer">
        <input
          className="searchInput"
          type="text"
          placeholder="Input your location"
          value={this.state.location}
          onChange={event => {
            this.setState({
              location: event.target.value,
            });
          }}
          onKeyPress={evt => {
            if (evt && evt.key === 'Enter') {
              this.props.onSearch(this.state.location);
              this.setState({ location: '' });
            }
          }}
        />
        <button
          className="searchBtn"
          onClick={() => {
            this.props.onSearch(this.state.location);
            this.setState({ location: '' });
          }}
        >
          Search
        </button>
        <style jsx>
          {`
            .searchboxContainer {
              width: 600px;
              display: flex;
              margin-left: 15px;
            }

            .searchInput {
              background-color: #ffffff;
              border: 0;
              border-radius: 5px;
              flex: 1;
              height: 30px;
              max-width: 600px;
              padding-left: 10px;
              transition: border 0.3s ease-out;
            }
            .searchInput:focus {
              border: 1px solid #cccccc;
            }
            .searchBtn {
              border: 0;
              border-radius: 4px;
              cursor: pointer;
              margin-left: 10px;
              height: 30px;
            }
          `}
        </style>
      </div>
    );
  }
}

export default Searchbox;
