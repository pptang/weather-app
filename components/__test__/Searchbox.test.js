import renderer from 'react-test-renderer';
import React from 'react';

import Searchbox from '../Searchbox';

describe('Pages With Snapshot Testing', () => {
  it('snapshot Searchbox Component', () => {
    const component = renderer.create(<Searchbox />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
