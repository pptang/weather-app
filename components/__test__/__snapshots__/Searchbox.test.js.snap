// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot Searchbox Component 1`] = `
<div
  className="jsx-2958326499 searchboxContainer"
>
  <input
    className="jsx-2958326499 searchInput"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Input your location"
    type="text"
    value=""
  />
  <button
    className="jsx-2958326499 searchBtn"
    onClick={[Function]}
  >
    Search
  </button>
</div>
`;
